### PR TITLE
OM-456: python3 compatibility

### PR DESCRIPTION
**Pull request by lesliegearhart**

### CODE REVIEW

File: onionconfig/admin/views.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 from onionconfig.admin.helpers import create_layer_list
-from django.contrib import messages
+from onionconfig.config import get_config, INVALID_CONFIG_FILES
+from onionconfig.signals import onion_config_updated
+
 
 def view_status(request):
     
-    if request.GET.get('reload', None)!=None:
+    if request.GET.get('reload', None) != None:
```
Comment (RevolutionTech) [inline onionconfig/admin/views.py]: nit: Use `is not None` instead, since `None` is a singleton

File: onionconfig/admin/views.py [hunk 2]
```diff
@@ -19,18 +21,16 @@ def view_status(request):
...
-    
-    print repr(filters)
-    filters = dict((k, v) for k,v in filters.items() if len(v))
-    print repr(filters)
 
-    context = {'filter_form': filter_form,
-               'view': get_config(path, **filters),
-               'layers': create_layer_list(filters)
-               }
+    filters = dict((k, v) for k, v in filters.items() if len(v))
```
Comment (RevolutionTech) [inline onionconfig/admin/views.py]: Python 2.7+ supports the following syntax:
```python
filters = {k: v for k, v in filters.items() if len(v)}
```

File: onionconfig/config.py [hunk 3]
```diff
@@ -58,24 +62,26 @@ def update(self, module_name):
...
             dir_ = os.path.dirname(dir_)
         self.directory = dir_
 
+
 class Layer(object):
     '''
     Stores a set of settings for a filtering
     '''
     def __init__(self, fname):
+
```
Comment (RevolutionTech) [inline onionconfig/config.py]: nit: why add this line?

File: onionconfig/config.py [hunk 4]
```diff
@@ -58,24 +62,26 @@ def update(self, module_name):
...
         data = eval(open(fname, "rb").read(), config.context)
-        #data = simplejson.load(open(fname, "rb"))
         assert isinstance(data, dict)
         filters = Layer._normalize_filters(data.pop("__filter", None))
-        self.priority = data.pop("__priority", None) or max(sum([config.dimensions[dim].priority_class for dim in filter_.keys()]) for filter_ in filters)
+        self.priority = data.pop("__priority", None) or max(
```
Comment (RevolutionTech) [inline onionconfig/config.py]: A lot is going on here. Consider breaking it up over multiple lines, or if you want to have it all in one line, here is how I would do it:
```python
self.priority = (
    data.pop("__priority")
    or max(
        sum([
            config.dimensions[dim].priority_class for dim in filter_.keys()
        ]) for filter_ in filters
    )
)
```

File: onionconfig/config.py [hunk 5]
```diff
@@ -84,41 +90,37 @@ def __init__(self, fname):
...
+        if data.get("__priority", 1) <= 0:
+            raise ValueError("Priority must be a positive integer")
         for dim in set([key for filter_ in Layer._preprocess_filter(data.get("__filter")) for key in filter_.keys()]):
-            if not config.dimensions.has_key(dim):
-                raise ValueError, "Unknown dimension used"
+            if dim not in config.dimensions:
+                raise ValueError("Unknown dimension used")
 
     @staticmethod
     def _normalize_filters(filters):
```
Comment (RevolutionTech) [inline onionconfig/config.py:100]: This method isn't doing the same thing anymore. Is that okay?

File: onionconfig/config.py [hunk 6]
```diff
@@ -84,41 +90,37 @@ def __init__(self, fname):
...
         # TODO(vhermecz): order expansions to the right order
         for expansion in config.expansions:
             new_filters = []
             for filter_ in filters:
-                if filter_.has_key(expansion.source_dimension_name):
+                if expansion.source_dimension_name in filter_:
                     target_values = []
                     for value in filter_[expansion.source_dimension_name]:
-                        target_values.extend([normalize(expansion.target_dimension_name, target_value) for target_value in expansion.expand(value)])
+                        target_values.extend(
```
Comment (RevolutionTech) [inline onionconfig/config.py]: Spacing could be improved:
```python
target_values.extend([
    normalize(expansion.target_dimension_name, target_value)
    for target_value in expansion.expand(value)
])
```

File: onionconfig/config.py [hunk 7]
```diff
@@ -170,59 +174,61 @@ def get_layers(directory=None):
             INVALID_CONFIG_FILES.append(fname)
             traceback.print_exc()
             logger.error("Invalid configuration layer")
-    res = [item[1] for item in sorted((config.get_priority(), config) for config in res)]
-    res = list(reversed(res))
+
+    res.sort(key=lambda x: x.get_priority(), reverse=True)
```
Comment (RevolutionTech) [inline onionconfig/config.py:184]: Nice, this is a great simplification! ✨

File: onionconfig/metaconfig.py [hunk 8]
```diff
@@ -33,15 +35,15 @@ class BaseDimension(object):
     priority_class = 0
     valueset = []
     
-    def __init__(self, name, label=name, description=None, priority_class=0, valueset = list()):
+    def __init__(self, name, label=name, description=None, priority_class=0, valueset=list()):
```
Comment (RevolutionTech) [inline onionconfig/metaconfig.py:38]: Having `list()` here is a bug. This instance will be shared across all instances.

**Review by RevolutionTech — APPROVED** [review]
LGTM ✨, though just like the other PR FWIW I didn't evaluate all of the code in this project to make sure it's all compatible with Python 3.

File: onionconfig/config.py (hunk 5 above)
Comment (lesliegearhart) [reply onionconfig/config.py:100]: Yeah, the results of `get_config` and `get_layers` were the same but I reverted the change because it's not necessary.

**Comment by alysivji** [comment]
@lesliegearhart is there anything blocking us from merging this PR in?

File: onionconfig/metaconfig.py (hunk 8 above)
Comment (alysivji) [reply onionconfig/metaconfig.py]: Fixed this

Comment (lesliegearhart) [reply onionconfig/metaconfig.py:38]: Unfortunately, `valueset` as a list shared across all `BaseDimension` instances is the desired/expected behavior where the library is used downstream. Added this as a known issue but leaving bug unfixed for now.

**Comment by lesliegearhart** [comment]
> @lesliegearhart is there anything blocking us from merging this PR in?

No, this is good to release

**Pull request merged**